### PR TITLE
add small benchmark to determine the performance of builtin functions

### PR DESCRIPTION
**Pull request by stlankes**

### CODE REVIEW

**Comment by stlankes** [comment]
I try to call directly builtin functions like [this benchmark](https://github.com/rust-lang/compiler-builtins/blob/master/testcrate/benches/mem.rs). Unfortunately, I am not to include directly the crate.

**Comment by stlankes** [comment]
@jbreitbart I added a benchmark, which directly calls `memcpy` and `memset`. Currently, I use the version of `compiler_builtin`. In the meantime, I am able to use the [assembly version](https://github.com/rust-lang/compiler-builtins/blob/master/src/mem/x86_64.rs). The performance of these functions is on my test system good.

File: benches/micro/src/benches/mod.rs [hunk 1]
```diff
@@ -117,3 +126,101 @@ pub fn bench_sched_two_threads() -> Result<(), ()> {
...
+
+pub fn bench_mem() -> Result<(), ()> {
+	memcpy_builtin(4096);
+	memcpy_builtin(1048576);
+	memset_builtin(4096);
+	memset_builtin(1048576);
+	memcpy_rust(4096);
+	memcpy_rust(1048576);
+	memset_rust(4096);
+	memset_rust(1048576);
```
Comment (jbreitbart) [inline benches/micro/src/benches/mod.rs:223]: 4KB and 1MB all fit into the cache of modern CPUs. Ideally we would also benchmark with a size that is twice the size of the last level cache of the system.

**Review by jbreitbart — APPROVED** [review]
We should implement these with criterion, once https://github.com/hermitcore/rusty-hermit/issues/53 is implemented. As a result, I am also fine with leaving the testcase sizes as they are for now.

File: benches/micro/src/benches/mod.rs (hunk 1 above)
Comment (stlankes) [reply benches/micro/src/benches/mod.rs]: Ok, I also also add a benchmark with a larger array size.

**Pull request merged**